### PR TITLE
New CLI package to start ethermint

### PR DESCRIPTION
**Pull request by adrianbrink**
This is the first package that is part of the new architecture.

I would love some early feedback. Right now there are a lot of todos and things will get moved around, but it shows the idea.

### CODE REVIEW

File: cli/utils.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"context"
+	"os/exec"
+	"time"
+)
+
+// InvokeTendermint invokes Tendermint in-process
+func InvokeTendermint(args ...string) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (odeke-em) [inline cli/utils.go:14]: I don't think you want to kill tendermint after 5 seconds :)

File: cli/development.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+// DevelopmentCommand is the entry point to create a private network.
+var DevelopmentCommand = &cobra.Command{
+	Use:   "development",
+	Short: "Start a local network for development.",
+	Run:   development,
+}
+
+func development(cmd *cobra.Command, args []string) {
+	// TODO: Launch ethermint and tendermint. Copy private keys.
```
Comment (zramsay) [inline cli/development.go:29]: isn't this done already in #245 ?

File: cli/development.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	// If none of the above is true throw an error and warn the user.
+	// Do the same for Tendermint.
+
+	emHome := viper.GetString("home")
+
+	// If the home directory does not exist then initialise everything.
+	// If it does exist we start everything straight away.
+
+	// TODO: This should be covered in five functions. TendermintInitialise(),
+	// TendermintStart(), EthereumInitialise(), EthereumStart(), CreateAccounts()
```
Comment (zramsay) [inline cli/development.go:42]: `EthermintInitialise`? ditto `EthermintStart`

File: cli/development.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+				// TODO: Handle error
+			}
+		}
+
+	}
+
+	fmt.Println(emHome)
+}
+
+var developmentPrivateKeys = map[string][]byte{
```
Comment (zramsay) [inline cli/development.go:94]: can we have an `ethermint keys new` like basecoin does?

File: cli/development.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+)
+
+func init() {
+	DevelopmentCommand.Flags().String("gaslimit", "", `The maximum amount of gas per block.`)
+}
+
+// DevelopmentCommand is the entry point to create a private network.
+var DevelopmentCommand = &cobra.Command{
+	Use:   "development",
+	Short: "Start a local network for development.",
```
Comment (zramsay) [inline cli/development.go:24]: a `Long:` with a bunch of additional info would be helpful

File: cli/mainnet.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// MainNetworkCommand is the command to connect to the main
+// network.
+var MainNetworkCommand = &cobra.Command{
+	Use:   "mainnet",
+	Short: "Connect to the main network.",
```
Comment (zramsay) [inline cli/mainnet.go:11]: `Long:` Ideally, it's written to describe the steps the main function takes to connect & what else happens. Also: `Connect to the main ethermint network` for clarity.

File: cli/testnet.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// TestNetworkCommand is the entry point to connect an ethermint instance to a
+// test network
+var TestNetworkCommand = &cobra.Command{
+	Use:   "testnet",
+	Short: "Connect to the test network.",
```
Comment (zramsay) [inline cli/testnet.go:11]: Long:

**Pull request closed without merging**